### PR TITLE
Remove moment

### PR DESCRIPTION
**Pull request by jordanstephens**
`moment` is enormous, and we don't use it for anything useful in here, so let's say goodbye.

while doing this i found a bunch of fns duplicted in `src/Model.js` and in `src/decorators/attribute.js`. I can't find anyone importing `attribute` from `src/Model`, so I went ahead and deleted it.

### CODE REVIEW

File: src/Model.js [hunk 1]
```diff
@@ -823,7 +708,9 @@ class Model {
         if (DataType.name === 'Array' || DataType.name === 'Object') {
           attr = toJS(value)
         } else if (DataType.name === 'Date') {
-          attr = moment(value).toISOString()
+          attr = value instanceof Date
+            ? value
+            : new Date(Date.parse(value)).toISOString()
```
Comment (jtorreggiani) [inline src/Model.js:713]: If `value` is a moment object here will it automatically coerce it to a string that is parseable by `Date.parse`?

File: src/decorators/attributes.js [hunk 2]
```diff
@@ -36,7 +35,8 @@ function defaultValueForDescriptor (descriptor, DataType) {
   if (typeof descriptor.initializer === 'function') {
     const value = descriptor.initializer()
     if (DataType.name === 'Date') {
-      return moment(value).toDate()
+      if (value instanceof Date) return value
+      return new Date(Date.parse(value))
```
Comment (jtorreggiani) [inline src/decorators/attributes.js:39]: Same goes for here.

**Comment by jtorreggiani** [comment]
@jordanstephens, did you run this in `artemis-store` to see if things break?

**Comment by jordanstephens** [comment]
@jtorreggiani yep, tests pass just find in `artemis-store` without any changes over there. 👍  i haven't run it in portal yet though

File: src/decorators/attributes.js (hunk 2 above)
Comment (jordanstephens) [reply src/decorators/attributes.js:39]: it mostly works as-is, but converting between moment and native dates, we actually lose millisecond precision. i haven't tracked that down yet (and i'm not sure it's important), but i can do a `value._isAMomentObject` here as well to just pass it on thru untouched.

Comment (jtorreggiani) [reply src/decorators/attributes.js]: That might be a good defensive measure.

**Pull request merged**